Implement Config Flow Workday (#22907)
home-assistant/home-assistant.github.io: source/_integrations/workday.markdown

Check the [country list](https://github.com/dr-prodigy/python-holidays#available-countries) for available provinces (and other subdivisions, like states and territories) for each country.

{% include integrations/config_flow.md %}

Days are specified as follows: `mon`, `tue`, `wed`, `thu`, `fri`, `sat`, `sun`.
The keyword `holiday` is used for public holidays identified by the holidays module.

<div class='note warning'>

> Watch how the `holiday` keyword is used. Your first instinct might be to add it to the `exclude` configuration, thinking that it means to skip the holidays. Actually it means to exclude the days in the holidays list from the workdays. So, when you exclude `holiday` and a workday falls on that day, then that workday is excluded, and the sensor will be **off**. If you want every workday flagged with no regard to holidays, make sure that there is something in your `Excludes` configuration _other_ than `holiday`.

Reviewer: ```suggestion
Watch how the `holiday` keyword is used. Your first instinct might be adding it to the `exclude` configuration, thinking it means skipping the holidays. It means to exclude the days in the holiday list from the workdays. So, when you exclude `holiday` and a workday falls on that day, that workday is excluded, and the sensor will be **off**. If you want every workday flagged with no regard to holidays, ensure that there is something in your `Excludes` configuration _other_ than `holiday`.
```

</div>

## Specific field information

Country code must be given according to [holidays](https://pypi.org/project/holidays/) notation.

Subdivision code must be given according to [holidays](https://pypi.org/project/holidays/) notation.

> Add holidays will only take dates formatted with YYYY-MM-DD.

Reviewer: ```suggestion
Add holidays will only take dates formatted with `YYYY-MM-DD`.
```

> Remove holidays will take dates formatted with YYYY-MM-DD or partial of name, as example `christmas` will exclude `Christmas Day`.

Reviewer: ```suggestion
Remove holidays will take dates formatted with `YYYY-MM-DD` or partial of name, for example, `christmas` will exclude `Christmas Day`.
```
